src/BasePaintThemeRegistry.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract BasePaintThemeRegistry is Ownable {
+    struct ThemeData {
+        string theme;
+        string[] palette;
+        uint256 size;
+    }
+
+    mapping(uint256 => ThemeData) private themes;
+    uint256 private nextThemeId = 1;
+
+    event ThemeUpdated(uint256 indexed themeId, string theme, string[] palette, uint256 size);
+
+    constructor() Ownable(msg.sender) {}
+
+    function setThemeData(string memory _theme, string[] memory _palette, uint256 _size)
+        public
+        onlyOwner
+        returns (uint256)
+    {
+        uint256 themeId = nextThemeId;
+        themes[themeId] = ThemeData(_theme, _palette, _size);
+        emit ThemeUpdated(themeId, _theme, _palette, _size);
+        nextThemeId++;
```

Comment:
I don't think this restriction makes sense. We should allow the owner to set any day's theme, and we could update things retroactively